### ISSUE TITLE
Preventing Linode deletion

### ISSUE THREAD

**Issue by cahebebe** [linode/manager#7092]
Please take in mind create an option when I could lock my Linone(s) VM when accidentally try to "delete" it, because it's too easy remove and lost data, would be extended to the another services.

Regards.

**Comment by nerdbaggy** [comment]
Would love this feature as well

**Comment by evanspy1** [comment]
+1, AWS has had this for a white, very helpful

**Comment by Jskobos** [comment]
Thanks for this feedback, sorry we missed this issue. We've been planning to make deleting a Linode type-to-confirm, which should make things a bit safer. I'll pass on the feedback about actually locking a Linode as well to see if it's something we can support.

**Comment by nerdbaggy** [comment]
It would be great to have the delete protection on API calls as well.

**Comment by mr-northwatch** [comment]
Yes, API protection would also be helpful.  Maybe a flag or secondary request?

**Comment by BAMLABS-AU** [comment]
+1 - I deleted one the other day and it was totally by accident. Lucky i had an offline backup as it deleted everything including the backups :(

**Comment by tsmgeek** [comment]
@bnussman that is protection after the fact, ie you want to delete it but then realise you may need to get some data
Deletion prevention should be there to also protect against mistaken actions.

As a side note the deletion protection should be an API permission, that way you can prevent say terraform from deleting anything, if protected. Or allow it to be created and protect, but after requires correct rights to alter/remove them.

This protection should also exist for Buckets/Volumes/NB etc.

**Comment by aaronheath** [comment]
How this isn't a feature yet is beyond me. Would make those coming from AWS that are used to instance protection feel (and be) much more secure.

This has got to be low hanging fruit and trivial to implement.

**Comment by dlong500** [comment]
@Jskobos It looks like you don't work for Linode anymore.  Any chance you could email a connection to look into this?  Really surprised this hasn't been addressed yet.  It's literally three clicks and your linode is gone without so much as a key typed.  I'm not as concerned about accidentally deleting it myself (though with a spazzy mouse it's entirely possible something horrible could happen).  I'm more concerned with a kid who might run into the home office when I'm getting coffee or something.

Deleting a linode should ideally require you to type a password, not just a confirmation phrase.  Having an option to enable  a 10-minute or so window during which a text alert is sent with the option of cancelling the operation would be icing on the cake.

**Comment by bnussman-akamai** [comment]
Hey all!

I know this is like 5 years later, but I'm seeing some movement on resource locking internally so stay tuned!
